src/logging.c

```diff
@@ -78,33 +87,40 @@ rcutils_ret_t rcutils_logging_initialize_with_allocator(rcutils_allocator_t allo
       g_rcutils_logging_output_format_string[chars_to_copy] = '\0';
     } else {
       if (NULL != ret_str) {
-        fprintf(
-          stderr,
-          "Failed to get output format from env. variable: %s. Using default output format.\n",
-          ret_str);
-        ret = RCUTILS_RET_INVALID_ARGUMENT;
+        if (rcutils_error_is_set()) {
+          fprintf(stderr, "Overwriting error message: %s\n", rcutils_get_error_string_safe());
```

Comment:
The `RCUTILS_SET_ERROR_MSG` does this already when you override an error state without clearing it.

Reply:
Yeah but it gives a help-like output suggesting you should reset the error message, so to me it seems more user friendly to do it manually where possible.

Reply:
I'm not sure I understood this bit, do you mean that places invoking `RCUTILS_SET_ERROR_MSG` should add this print when they need to reset the error message because the one printed by `RCUTILS_SET_ERROR_MSG` is not clear ?
Would making `RCUTILS_SET_ERROR_MSG` overwrite message clearer solve this without having to add printfs manually wherever we reset the error message?

Reply:
> Would making RCUTILS_SET_ERROR_MSG overwrite message clearer solve this without having to add printfs manually wherever we reset the error message?

we could change it, yeah, but to me the issue is that it's a little _too_ helpful at the moment, so it would actually have to be made what can be considered _less_ clear: https://github.com/ros2/rcutils/blob/f9a0537e4008d9c12e32dbf51857e7a30a8f64e0/src/error_handling.c#L157 To me it looks like a bug if that gets output, so I avoided it since in this case overwriting the error message is intentional.

Reply:
since the previous error now causes a return (ac66eced2f3c381a887987ddb65af2d0b71017a4), this code has gone away naturally :smile: